coast/config/config_structure.py

```diff
@@ -0,0 +1,108 @@
+"""Classes defining config structure."""
+from dataclasses import dataclass
+from enum import Enum, unique
+
+
+@unique
+class ConfigTypes(Enum):
+    """Enum class containing the valid types for config files."""
+    GRIDDED = "gridded"
+    INDEXED = "indexed"
+
+
+class ConfigKeys():
+    """Class of constants representing valid keys within configuriation json."""
+    TYPE="type"
+    DIMENSIONALITY="dimensionality"
+    GRIDREF="grid_ref"
+    PROC_FLAGS="processing_flags"
+    DATASET="dataset"
+    DOMAIN="domain"
+    CODEPROCESSING="static_variables"
+    DIM_MAP="dimension_map"
+    VAR_MAP="variable_map"
+    CHUNKS="chunks"
+    NO_GR_VAR = "not_grid_vars"
+    COO_VAR="coord_vars"
+    DEL_VAR="delete_vars"
+
+
+@dataclass(frozen=True)
+class DataFile():
+    """General parent dataclass for holding common config attributes of datafiles.
+    
+    Args:
+        variable_map (dict): dict containing mapping for variable names.
+        dimension_map (dict): dict containing mapping for dimension names.
+    """
+    variable_map: dict
+    dimension_map: dict
+
+
+@dataclass(frozen=True)
+class CodeProcessing:
+    """
+    Dataclass holding config attributes for static variables that might not need changing between model runs
+    """
+    not_grid_variables: list
+    coord_variables: list
+    delete_variables: list
+
+
+@dataclass(frozen=True)
+class Dataset(DataFile):
+    """
+    Dataclass holding config attributes for Dataset datafiles. Extends DataFile.
```

Comment:
We should be consistent with docstrings for dataclasses.

Reply:
Sorry, I meant that the above dataclass (DataFile) mentions its parameters in the docstring, this one does not.